src/Common/tests/System/PlatformDetection.cs

```diff
@@ -44,47 +45,43 @@ public static partial class PlatformDetection
 
         public static bool IsNetfx462OrNewer()
         {
-            if (!IsFullFramework)
-            {
-                return false;
-            }
-
             Version net462 = new Version(4, 6, 2);
             Version runningVersion = GetFrameworkVersion();
             return runningVersion != null && runningVersion >= net462;
         }
 
         public static bool IsNetfx470OrNewer()
         {
-            if (!IsFullFramework)
-            {
-                return false;
-            }
-
             Version net470 = new Version(4, 7, 0);
             Version runningVersion = GetFrameworkVersion();
             return runningVersion != null && runningVersion >= net470;
         }
 
         public static Version GetFrameworkVersion()
         {
-            string[] descriptionArray = RuntimeInformation.FrameworkDescription.Split(' ');
-            if (descriptionArray.Length < 3)
-            {
+            if (!IsFullFramework)
                 return null;
-            }
-                
-            string runningVersion = descriptionArray[2];
 
-            // we could get a version with build number > 1 e.g 4.6.1375 but we only want to have 4.6.1
-            // so that we get the actual Framework Version
-            if (runningVersion.Length > 5)
+            using (RegistryKey ndpKey = RegistryKey.OpenRemoteBaseKey(RegistryHive.LocalMachine, "").OpenSubKey(@"SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full"))
```

Comment:
could you please add a comment including the link https://msdn.microsoft.com/en-us/library/hh925568(v=vs.110).aspx for more info. Also please tell this method will need to be modified every time we release a new version of the full framework.